Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -17,15 +17,50 @@ import ArcGIS
 
 /// An object that encapsulates state about a preplanned map.
 @MainActor
-class PreplannedMapModel: ObservableObject, Identifiable {
+public class PreplannedMapModel: ObservableObject, Identifiable {
     /// The preplanned map area.
     let preplannedMapArea: any PreplannedMapAreaProtocol
     
+    /// The task to use to take the area offline.
+    private let offlineMapTask: OfflineMapTask
+    
+    /// The ID of the web map.
+    private let portalItemID: Item.ID
+    
+    /// The ID of the preplanned map area.
+    private let preplannedMapAreaID: Item.ID
+    
+    /// The mobile map package for the preplanned map area.
+    private(set) var mobileMapPackage: MobileMapPackage?
+    
+    /// The currently running download job.
+    @Published private(set) var job: DownloadPreplannedOfflineMapJob?
+    
     /// The combined status of the preplanned map area.
     @Published private(set) var status: Status = .notLoaded
     
-    init(preplannedMapArea: PreplannedMapAreaProtocol) {
-        self.preplannedMapArea = preplannedMapArea
+    /// A Boolean value indicating if a user notification should be shown when a job completes.
+    let showsUserNotificationOnCompletion: Bool
+    
+    init(
+        offlineMapTask: OfflineMapTask,
+        mapArea: PreplannedMapAreaProtocol,
+        portalItemID: Item.ID,
+        preplannedMapAreaID: Item.ID,
```

Comment:
```swift
        portalItemID: PortalItem.ID,
        preplannedMapAreaID: PortalItem.ID,
```
Up to you.

In terms of the preplanned map area ID, I think they can be counted as "portal item ID" as well. For instance, if I access https://www.arcgis.com/home/item.html?id=20244977a6434c50afa9c5e1b6ab670c , which is the City Hall preplanned map area of the naperville webmap, even if it doesn't show the affiliation to the webmap, it is still accessible as a portal item.